### ISSUE TITLE
[MR / Accessibility] List of cards provided without list semantics

### ISSUE THREAD

**Issue by sarahhortonadhoc** [department-of-veterans-affairs/va.gov-team#128666]
### 📝 User story

As a screen reader user, I want lists to be provided using list elements, so that I can understand that I am navigating list content and know how many items are in the list, and navigate between list items.

### 📝 Problem statement

Cards in lists are provided using non-semantic `<div>` elements.

### 🌌 Supporting screenshots or videos

<img width="670" height="434" alt="Screenshot of a list of cards showing medical record details" src="https://github.com/user-attachments/assets/cb52d7e6-29d4-4d71-bbf9-f0d579cdd7be" />

### 👣 Reproduction steps

1. Navigate to the [Lab and test results list page](https://staging.va.gov/my-health/medical-records/labs-and-tests) (username: (mhvmark.test@id.me).
2. Use the ARC Toolkit browser extension to show lists.
3. Notice that there are no lists.

### 🔗 Point of contact

@sarahhortonadhoc

### ⚠️ Severity level

2, Serious. Should be fixed in 1-2 sprints post-launch.

#### Accessibility Standards
- [Guideline 1.3 Adaptable: Create content that can be presented in different ways (for example simpler layout) without losing information or structure.](https://www.w3.org/WAI/WCAG22/Understanding/adaptable)
  - [1.3.1 Info and Relationships (Level A)](https://www.w3.org/WAI/WCAG22/Understanding/info-and-relationships.html)

#### Experience Standards
- [Category Usability: User is able to easily know or learn how to use the interface to complete tasks.](https://design.va.gov/about/experience-standards/#category-usability)
  - [User can identify an element and its state](https://design.va.gov/about/experience-standards/#user-can-identify-an-element-and-its-state)

### 𝌣 Proposed solution or next steps

Use list elements (`<ul>` and `<li>`) to provide card lists (see [Medications list page](https://staging.va.gov/my-health/medications?page=1) for an example of a correct implementation).

### ✅ Acceptance criteria

- [ ] Semantics are reflected in Figma annotations
- [ ] The system displays the list of cards as an ordered list and cards as list items.
- [ ] The screen reader announces the list and list items, and the number of items in the list.

### ✅ Possible tasks

- [ ] Review, refine, and acknowledge feedback.
- [ ] Update ticket information (title, assignee, labels, project(s), etc.)
- [ ] Fix and/or document decisions made.
- [ ] Product manager to move ticket in their respective product board to 'QA/A11y Review' after reviewing documented decisions / validating fix.
- [ ] Accessibility specialist will close the ticket after reviewing documented decisions / validating the fix.
